src/main/java/de/tum/in/www1/artemis/service/TeamService.java

```diff
@@ -205,6 +224,76 @@ public Team save(Exercise exercise, Team team) {
         return teamRepository.findAllByExerciseId(destinationExercise.getId());
     }
 
+    /**
+     * Converts teams' students with only login or registration number to students on database
+     *
+     * @param course Course in which the users will be searched
+     * @param teams Teams that students are described only by login or visible registration number
+     * @return list of all teams that now have registered users
+     */
+    public List<Team> convertTeamsStudentsWithOnlyLoginOrRegistrationNumberToAlreadyRegisteredUsers(Course course, List<Team> teams) {
```

Comment:
This is a huge method and in my opinion has to be split into multiple smaller functions. 

As a first the the function converts via login or registration number so I'd break this down into:

```java
public List<Team> convertTeamsStudentsWithOnlyLoginToAlreadyRegisteredUsers(Course course, List<Team> teams);
public List<Team> convertTeamsStudentsWithRegistrationNumberToAlreadyRegisteredUsers(Course course, List<Team> teams);
```

But the method name is still really long so maybe you can try to split it even more.